--- a/rust/private/rustc.bzl
+++ b/rust/private/rustc.bzl
@@ -1981,7 +1981,7 @@ def rustc_compile(
         use_json_output = bool(build_metadata) or bool(rustc_output) or bool(rustc_rmeta_output),
         skip_expanding_rustc_env = skip_expanding_rustc_env,
         require_explicit_unstable_features = require_explicit_unstable_features,
-        always_use_param_file = rust_toolchain._bootstrapping,
+        always_use_param_file = rust_toolchain._bootstrapping or rust_toolchain.target_os == "windows",
         inject_allow_features_guardrail = inject_allow_features_guardrail,
         allowed_unstable_rust_features = allowed_unstable_rust_features,
         runtime_libs = runtime_libs,
