--- a/rust/private/rustc.bzl
+++ b/rust/private/rustc.bzl
@@ -470,0 +471,8 @@
+                elif processed_arg == "-L" and i + 1 < len(link_args):
+                    path = link_args[i + 1]
+                    if ld_is_direct_driver and toolchain.target_os == "windows" and toolchain.target_abi == "msvc":
+                        skip_next = True
+                        continue
+                    filtered_args.extend([processed_arg, path])
+                    skip_next = True
+
@@ -471,0 +480,2 @@
+                    if ld_is_direct_driver and toolchain.target_os == "windows" and toolchain.target_abi == "msvc":
+                        continue
@@ -475,0 +486,4 @@
+                    if ld_is_direct_driver and toolchain.target_os == "windows" and toolchain.target_abi == "msvc":
+                        if processed_arg == "--sysroot" and i + 1 < len(link_args):
+                            skip_next = True
+                        continue
@@ -828,0 +843 @@
+            toolchain.rust_std,
@@ -2945,0 +2961 @@
+        for_windows = False,
@@ -2947 +2963,2 @@
-        flavor_msvc = False):
+        flavor_msvc = False,
+        use_direct_driver = False):
@@ -2954,0 +2972 @@
+        for_windows (bool, optional): _description_. Defaults to False.
@@ -2956,0 +2975 @@
+        use_direct_driver (bool, optional): _description_. Defaults to False.
@@ -2999,0 +3019,5 @@
+        if for_windows and flavor_msvc and use_direct_driver and not artifact.basename.endswith(".lib"):
+            return [
+                "-Clink-arg={}".format(artifact.path),
+            ]
+
@@ -3048 +3072 @@
-            ret.extend(portable_link_flags(lib, use_pic, ambiguous_libs, get_lib_name, flavor_msvc = flavor_msvc))
+            ret.extend(portable_link_flags(lib, use_pic, ambiguous_libs, get_lib_name, for_windows = True, flavor_msvc = flavor_msvc, use_direct_driver = use_direct_driver))
@@ -3260,5 +3284,12 @@
-                args.add_all(
-                    runtime_libs,
-                    map_each = get_lib_name,
-                    format_each = static_runtime_link_format,
-                )
+                if toolchain.target_os == "windows" and toolchain.target_abi == "msvc" and use_direct_link_driver:
+                    for runtime_lib in runtime_libs.to_list():
+                        if runtime_lib.basename.endswith(".lib"):
+                            args.add(get_lib_name(runtime_lib), format = static_runtime_link_format)
+                        else:
+                            args.add(runtime_lib, format = "--codegen=link-arg=%s")
+                else:
+                    args.add_all(
+                        runtime_libs,
+                        map_each = get_lib_name,
+                        format_each = static_runtime_link_format,
+                    )
