--- a/rs/private/annotations.bzl
+++ b/rs/private/annotations.bzl
@@ -2,6 +2,8 @@
         additive_build_file = None,
         additive_build_file_content = "",
         gen_build_script = "auto",
+        build_script_deps = [],
+        proc_macro_deps = [],
         build_script_data = [],
         build_script_data_select = {},
         build_script_env = {},
@@ -29,6 +31,8 @@
         additive_build_file = additive_build_file,
         additive_build_file_content = additive_build_file_content,
         gen_build_script = gen_build_script,
+        build_script_deps = build_script_deps,
+        proc_macro_deps = proc_macro_deps,
         build_script_data = build_script_data,
         build_script_data_select = build_script_data_select,
         build_script_env = build_script_env,
